design-documents/webp-support.md

```diff
@@ -0,0 +1,43 @@
+## Add support for the WebP image format.
+
+### Terms
+
+WebP is an image format employing both lossy and lossless compression.
+https://developers.google.com/speed/webp/
+
+### Overview
+
+Using WebP images should speed up a page load for customers.
+WebP offers 25 – 35% smaller file sizes in compare with JPEG with the same quality. ([WebP Compression Study](https://developers.google.com/speed/webp/docs/webp_study))
+
+[Supporting WebP by browsers](https://caniuse.com/#feat=webp)
+
+### Design
+
+On image resizing for each size, Magento can create one more image in WebP format.
+For prevention unexpected resource consumption it can be managed by configuration and by default be disabled.
+A web server will be responsible for the decision which file send.
+This decision will be made by availability the requested image with WebP extension and supporting WebP by the browser using the Accept header.
+
+#### Component Dependencies
+
+This functionality requires an asynchronous image resizing to avoid degradation on adding images to the gallery and resizing all images.
```

Comment:
How will the asynchronous image resizing be implemented? Is it going to be a separated async service that would be triggered by webserver to handle image requests?

Reply:
Asynchronous image resizing hasn't been implemented yet.
But in my opinion, when a resized image does not exist, it should be resized in a synchronous way. Asynchronous resizing should be used for adding an image to a media gallery and for all images resizing.